File: AudioQueue/audio_queue.py
class AudioQueue:
    def __init__(self, max_size):
        self.queue = []
        self.max_size = max_size

    def enqueue(self, item):
        if len(self.queue) < self.max_size:
            self.queue.append(item)
        else:
            print("Queue is full. Cannot enqueue more items.")

    def dequeue(self):
        if self.is_empty():
            print("Queue is empty. Cannot dequeue.")
        else:
            return self.queue.pop(0)

    def is_empty(self):
        return len(self.queue) == 0

    def is_full(self):
        return len(self.queue) == self.max_size

    def size(self):
        return len(self.queue)
    
    def empty_queue(self):
        self.queue = []  # Clears all items from the queue
    
    def __str__(self):
        return f"AudioQueue: {self.queue}"

File: src/app/helper/AudioQueue.ts
export class AudioQueue<T> {
    private queue: T[];
    private max_size: number;

    constructor(max_size: number) {
        this.queue = [];
        this.max_size = max_size;
    }

    enqueue(item: T): void {
        if (this.queue.length < this.max_size) {
            this.queue.push(item);
        } else {
            console.log("Queue is full. Cannot enqueue more items.");
        }
    }

    dequeue(): T | undefined {
        if (this.is_empty()) {
            console.log("Queue is empty. Cannot dequeue.");
            return undefined;
        } else {
            return this.queue.shift();
        }
    }

    is_empty(): boolean {
        return this.queue.length === 0;
    }

    is_full(): boolean {
        return this.queue.length === this.max_size;
    }

    size(): number {
        return this.queue.length;
    }

    empty_queue(): void {
        this.queue = [];
    }

    toString(): string {
        return `AudioQueue: ${JSON.stringify(this.queue)}`;
    }
}